lib/cipherstash/index.rb

```diff
@@ -23,16 +27,22 @@ def self.generate(id, settings, schema_versions)
         Exact.new(id, settings, schema_versions)
       when "range"
         Range.new(id, settings, schema_versions)
-      when "match"
-        Match.new(id, settings, schema_versions)
-      when "dynamic-match"
-        DynamicMatch.new(id, settings, schema_versions)
-      when "field-dynamic-match"
-        FieldDynamicMatch.new(id, settings, schema_versions)
+      when "match", "ore-match"
+        OreMatch.new(id, settings, schema_versions)
+      when "dynamic-match", "dynamic-ore-match"
+        DynamicOreMatch.new(id, settings, schema_versions)
+      when "field-dynamic-match", "field-dynamic-ore-match"
+        FieldDynamicOreMatch.new(id, settings, schema_versions)
       when "field-dynamic-exact"
         FieldDynamicExact.new(id, settings, schema_versions)
+      when "filter-match"
+        FilterMatch.new(id, settings, schema_versions)
+      when "dynamic-filter-match"
+        DynamicFilterMatch.new(id, settings, schema_versions)
+      when "field-dynamic-filter-match"
+        FieldDynamicFilterMatch.new(id, settings, schema_versions)
       else
-        raise Error::InvalidSchemaError, "Unknown index kind #{settings["mapping"]["kind"].inspect}"
+        raise Client::Error::InvalidSchemaError, "Unknown index kind #{settings["mapping"]["kind"].inspect}"
```

Comment:
Noticed that the error class was slightly off here while adding in the new index kinds. This was causing an "unitialized constant" error.

There are a few other spots like this in this file. Should we patch those up in a separate change @mpalmer?

Reply:
Yeah, probably best to it in a separate PR, for isolation.  There's likely to be a fair number of those sorts of problems squirreled away, because without the guiderails of tests, I'm a fairly mediocre programmer.

Reply:
Added a card to Notion for this one.